File: sample1/python/reader.py



"""
Protocol buffer reader in Python

author: Atsushi Sakai
"""

import addressbook_pb2


def main():
    """
    main func
    """
    print("start!!")

    address_book = addressbook_pb2.AddressBook()

    # Read the existing address book.
    # file_path = open("pbdata_py.dat", "rb")  # read data from python code
    # file_path = open("../cpp/pbdata_cpp.dat", "rb")  # read data from cpp code
    file_path = open("../julia/pbdata_julia.dat",
                     "rb")  # read data from cpp code

    address_book.ParseFromString(file_path.read())
    file_path.close()

    print(address_book)

    print("done!!")


if __name__ == '__main__':
    main()

File: sample1/cpp/reader.cpp
/**
 * Protocol buffer data reader in cpp
 *
 * @author Atsushi Sakai
 **/
#include <fstream>
#include <iostream>
#include <string>

#include "addressbook.pb.h"

using namespace std;

int main(void) {
  cout << "Start Protocol buffers reader sample" << endl;
  tutorial::AddressBook address_book;

  // fstream input("./pbdata_cpp.dat", ios::in | ios::binary);
  // fstream input("../python/pbdata_py.dat", ios::in | ios::binary);
  fstream input("../julia/pbdata_julia.dat", ios::in | ios::binary);
  address_book.ParseFromIstream(&input);

  // Human readable print
  cout << address_book.DebugString() << endl;

  cout << "Done" << endl;
}
